eth-bytecode-db/eth-bytecode-db/src/verification/handlers/mod.rs

```diff
@@ -338,33 +379,27 @@ async fn process_verifier_alliance_db_action(
                 anyhow::bail!("Both creation and runtime codes are nulls")
             }
 
-            // TODO: make transaction_hash input argument optional, and calculate it
-            //       as `keccak256(creation_code || runtime_code)` in that case
-            let transaction_hash = transaction_hash.to_vec();
+            let transaction_hash = derive_transaction_hash(
+                transaction_hash.clone(),
+                creation_code.clone(),
+                runtime_code.clone(),
+            )
+            .expect("At least one of creation or runtime codes must exist");
```

Comment:
I would remove this expect and bail on L379 and add `ok_or_else(|| anyhow::anyhow("Both creation and runtime codes are nulls"))?` to derive_transaction_hash